src/solvers/smt2/smt2_solver.cpp

```diff
@@ -81,39 +81,42 @@ void smt2_solvert::expand_function_applications(exprt &expr)
   {
     auto &app=to_function_application_expr(expr);
```

Comment:
Commit message: "applying functions that aren't symbols" does not match "A properly formed Git commit subject line should always be able to complete the following sentence: If applied, this commit will _your subject line here_" (from https://chris.beams.io/posts/git-commit/)

Also: s/lamba/lambda/

Reply:
extended